recipes/libuuid/all/patches/0001-Check-for-HAVE_SYS_SYSCALL_H.patch

```diff
@@ -0,0 +1,32 @@
+From 32dba1f82848e2bf4e2546dac06296fa7b387dff Mon Sep 17 00:00:00 2001
+From: Rob Boehne <robb@datalogics.com>
+Date: Mon, 16 Nov 2020 09:06:42 -0600
+Subject: [PATCH] Check for HAVE_SYS_SYSCALL_H
+
+Fix a compilation error on systems without sys/syscall.h by applying the same checks used elsewhere.
+---
+ randutils.c | 5 +++++
+ 1 file changed, 5 insertions(+)
+
+diff --git a/randutils.c b/randutils.c
+index 80893d3..8bf3af6 100644
+--- a/randutils.c
++++ b/randutils.c
+@@ -11,9 +11,14 @@
+ #include <fcntl.h>
+ #include <stdlib.h>
+ #include <string.h>
++
++#if defined(HAVE_SYS_TIME_H)
+ #include <sys/time.h>
++#endif
+ 
++#if defined(__linux__) && defined(HAVE_SYS_SYSCALL_H)
```

Comment:
I see you've omitted the patch adding the test for `sys/syscall.h` in `configure.ac`.
Won't this make it stop building for linux?

Reply:
It doesn't - the check is done implicitly, but ideally the configury would also be updated.
It becomes complicated to add that properly because it will change the contents of the generated files, in ways that won't really be helpful.  This project seems to not get too much attention, so if it ever does get merged the next version will not need the patch.  I've also verified the Linux build does not break.

Reply:
is `__linux__` needed here? it may omit other operation systems (e.g. BSD)

Reply:
I put it here only because it appeared in the only other place this header is included.  IMO it isn't needed, and `sys/syscalls.h` (as I recall) is 100% Linux-specific so it's really just redundant.   I don't have a BSD system handy to check though.